### PR TITLE
Implement rate limiting

### PR DESCRIPTION
**Pull request by phhai-hcmut**
Rate limit API only in production, not in development or test.

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -19,16 +23,27 @@
...
 )
 
+limiter = Limiter(
+    key_func=get_remote_address,
+    default_limits=[settings.RATE_LIMIT],
+    enabled=settings.PYTHON_ENV == "production",
+)
+app.state.limiter = limiter
+app.add_exception_handler(RateLimitExceeded, _rate_limit_exceeded_handler)
+app.add_middleware(SlowAPIMiddleware)
```
Comment (produdez) [inline app/main.py:33]: Can you explain a bit abt how this works? Im lazy to go read `slowapi` document

**Review by produdez — COMMENTED** [review]
Please add some test or sth to make sure that the `Limiter` actually works.

File: app/main.py (hunk 1 above)
Comment (phhai-hcmut) [reply app/main.py:33]: Create an app-wide instance of python `collections.Counter` to store client connected, with key got by `get_remote_address()` and value is number of request in the time window. If value exceed `settings.RATE_LIMIT`, return 429 error.

**Comment by phhai-hcmut** [comment]
We don't have to write test for everything :)

**Comment by produdez** [comment]
We're running out of thing to do :v. Also I think if we can test this one it's pretty nice :3

**Comment by phhai-hcmut** [comment]
How do we even test this? Fake IP address?

**Comment by produdez** [comment]
I really dont know :v, i do know that creating a new instance of `TestClient` will mimic a different client.

**Comment by produdez** [comment]
Maybe reduce the `ratelimit` to some very low number in the test so that we can actually exceed it?

**Comment by phhai-hcmut** [comment]
Why do we need to test functionality of a library? Sound pretty pointless to me.

**Comment by produdez** [comment]
If you're sure it works than im good :v, i just dont know if it works 😄. You can merge if it's enough.

**Pull request merged**